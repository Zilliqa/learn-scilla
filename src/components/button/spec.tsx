import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { FaCheck } from 'react-icons/fa';
import Button from '.';

describe('Button tests', () => {
  const baseComponent = (props) => (
    <Button
      type={props.type}
      text={'ButtonText'}
      disabled={false}
      onClick={() => console.log('click')}
      ariaLabel={'Test Button'}
      before={<FaCheck />}
      size={'md'}
    />
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent({ type: 'secondary' })).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders primary button without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ type: 'primary' }), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders secondary button without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ type: 'secondary' }), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders warning button crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ type: 'warning' }), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders danger button without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ type: 'danger' }), div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });
});
