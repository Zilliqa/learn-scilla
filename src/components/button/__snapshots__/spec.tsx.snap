// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests basic tests matches the snapshot 1`] = `
<button
  aria-label="Test Button"
  className="btn btn-md type-secondary "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
    />
  </svg>
   
  ButtonText
   
</button>
`;
